### ISSUE TITLE
Documentation references 'gradle-site-plugin' which builds with Gradle 5.4.1

### ISSUE THREAD

**Issue by bigdaz** [gradle/gradle#20236]
The various "Gradle on CI" chapters in the User Guide each use https://github.com/gradle/gradle-site-plugin as an example project to build and test. eg:
- https://docs.gradle.org/current/userguide/jenkins.html
- https://docs.gradle.org/current/userguide/teamcity.html
- https://docs.gradle.org/current/userguide/travis-ci.html

This repository is very out of date, and should be modernized:
- Update to use latest version of Gradle (currently uses Gradle 5.4.1)
- Use the new test-suite functionality to define the 'integrationTest' task
- Apply the `com.gradle.common-custom-user-data-gradle-plugin` to add tags/labels to the generated build scan (instead of defining custom build-scan customizations).
- Apply the latest version of `com.gradle.plugin-publish` plugin
- ... various other things could be improved ...

Alternatively, a different, simpler, sample project could be used for documenting CI systems with Gradle.

### Expected Behavior

Projects referenced by documentation should build with a newer version of Gradle.

### Current Behavior

The project linked from the CI chapters builds with Gradle 5.4.1

**Comment by big-guy** [comment]
I don't think we should update that repo at all. We removed all of the references to it from the other guides we had. I guess we didn't update the ones for TeamCity/Travis/Jenkins.

We should update the first part of these guides to use gradle init and delete the references to gradle-site-plugin.

**Comment by bigdaz** [comment]
> We should update the first part of these guides to use gradle init and delete the references to gradle-site-plugin.

This is a good point @big-guy. I've updated the new GitHub Actions guide to use `gradle init` to initialize a Gradle project.
1971b52a1448f01ad2ec07e4ff713f2f687a826a
I have not updated the other guides, but it may be possible to extract the content out of the new guide and share between all of the CI guides.

**Comment by lkasso** [comment]
These pages are now hosted on the community site. 
The issue has been moved to: https://github.com/gradle/community/issues/188
